src/main/java/net/dv8tion/jda/api/interactions/CustomIdInteraction.java

```diff
@@ -0,0 +1,29 @@
+package net.dv8tion.jda.api.interactions;
+
+import net.dv8tion.jda.api.interactions.components.ComponentInteraction;
+import net.dv8tion.jda.api.interactions.modals.ModalInteraction;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Interactions which have a custom ID, namely {@link ComponentInteraction} and {@link  ModalInteraction}.
+ *
+ * <br>This id does not have to be numerical.
+ *
+ */
+public interface CustomIdInteraction
```

Comment:
I would make this a marker interface, we usually prefix those with an `I` (I know it's odd, but we should keep that consistent here)

```suggestion
public interface ICustomIdInteraction
```